### PR TITLE
Fix tests reverting typescript version to 3.9.6

### PR DESCRIPTION
**Pull request by JSReds**
Revert typescript version to 3.9.6 to fix several tests errors

Change-type: patch
Signed-off-by: Andrea Rosci <andrear@balena.io>

---
##### Contributor checklist

- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

File: src/components/Renderer/index.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const getValue = (
 	schema?: JSONSchema,
 	uiSchema?: UiSchema,
 ) => {
-	const calculatedValue = uiSchema?.['ui:value'];
+	const calculatedValue = uiSchema?.['ui:value'] ?? value;
```
Comment (thgreasi) [inline src/components/Renderer/index.tsx:28]: The previous refactor on this didn't actually change the behavior at all.
In my attempts, the tests passed even w/o this.

Comment (Page-) [reply src/components/Renderer/index.tsx:28]: Yeah this is wrong as it treats null and undefined the same @JSReds, you can see below we specifically handle null vs undefined separately and do the defaulting to value as need be

File: src/components/Renderer/index.tsx [hunk 2]
```diff
@@ -166,7 +166,7 @@ const RendererBase = ({
 	);
 	const Widget = getWidget(
 		processedValue,
-		schema.format,
+		schema?.format,
```
Comment (Page-) [inline src/components/Renderer/index.tsx:169]: Can you update `ThemedRendererProps` to denote that `schema` is actually optional and can be undefined please

File: package.json [hunk 3]
```diff
@@ -83,7 +83,7 @@
     "strip-css-comments": "^3.0.0",
     "ts-jest": "^26.5.4",
     "ts-node": "^9.1.1",
-    "typescript": "~4.1.5",
+    "typescript": "^3.9.6",
```
Comment (Page-) [inline package.json:86]: I'd prefer to make things work with the newer typescript rather than pin to 3.x if possible

**Review by Page- — CHANGES_REQUESTED** [review]
Request changes for the null vs undefined issue

File: src/components/Renderer/index.tsx (hunk 2 above)
Comment (JSReds) [reply src/components/Renderer/index.tsx:169]: Just realized that this is not making much sense, I'm going to check the examples and fix them

Comment (thgreasi) [reply src/components/Renderer/index.tsx:169]: This seems to only happen for the examples about materialized and invalid fields.
See: https://github.com/balena-io-modules/rendition/blob/2becdac6dfc642f6e02ec7c5fa5b5c90343c3823/src/components/Renderer/examples.ts#L116-L126
See: https://github.com/balena-io-modules/rendition/blob/2becdac6dfc642f6e02ec7c5fa5b5c90343c3823/src/components/Renderer/examples.ts#L341-L366

Imo the typings are correct, it's just that atm we try to use the Renderer recursively before checking whether the resulting schema is valid. What'st the correct thing to do for the materialized field though is harder to answer.

File: package.json (hunk 3 above)
Comment (thgreasi) [reply package.json:86]: I agree on principle @Page- , but the TS bump has caused a weird `window.matchMedia is undefined` error in storybook, so it's not a TS compilation issue thing.
I do see some profit on getting master to have its tests passing asap and fix the storybook issue in a follow-up PR.

**Review by thgreasi — APPROVED** [review]

**Review by Page- — APPROVED** [review]

**Pull request merged**